### PR TITLE
[SR SQL Planner Summer Camp] Support anltr4 for showAlterStmt

### PR DESCRIPTION
**Pull request by xlwh**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7938

### CODE REVIEW

**Comment by xlwh** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 45 / 55 (81.82%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|14|18|77.78%|
|:large_blue_circle:|com/starrocks/analysis/ShowAlterStmt.java|5|6|83.33%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowAlterStmtAnalyzer.java|20|24|83.33%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|2|2|100.00%|

**Comment by xlwh** [comment]
run starrocks_fe_unittest

**Comment by xlwh** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 1]
```diff
@@ -2423,6 +2423,11 @@ show_param ::=
     {:
         RESULT = new ShowExportStmt(db, parser.where, orderByClause, limitClause);
     :}
+    /* Show delete statement */
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/cup/sql_parser.cup:2426]: has conflicts?

File: fe/fe-core/src/main/java/com/starrocks/analysis/ShowAlterStmt.java [hunk 2]
```diff
@@ -149,9 +156,6 @@ private void analyzeSubPredicate(Expr subExpr) throws AnalysisException {
     public void analyze(Analyzer analyzer) throws AnalysisException, UserException {
         //first analyze 
         analyzeSyntax(analyzer);
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/analysis/ShowAlterStmt.java:158]: The functionality of analyzeSyntax has not been moved into the new Analyzer. This part of the code will be lost in the new Analyzer

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 41 / 50 (82.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|12|16|75.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowAlterStmt.java|5|6|83.33%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|20|23|86.96%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|

**Comment by xlwh** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 66 / 122 (54.10%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowAlterStmtAnalyzer.java|37|87|42.53%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|12|16|75.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowAlterStmt.java|11|12|91.67%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|2|2|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 96 / 127 (75.59%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowAlterStmtAnalyzer.java|60|87|68.97%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowAlterStmt.java|14|16|87.50%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|16|17|94.12%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|2|2|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 103 / 127 (81.10%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowAlterStmtAnalyzer.java|67|87|77.01%|
|:large_blue_circle:|com/starrocks/analysis/ShowAlterStmt.java|14|16|87.50%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|16|17|94.12%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|2|2|100.00%|

**Comment by xlwh** [comment]
@HangyuanLiu  Thank you for the review, the related comments and some test problems have been fixed, please help and take a look again, thank you!

**Comment by xlwh** [comment]
run starrocks_fe_unittest

**Comment by xlwh** [comment]
run starrocks_fe_unittest

**Review by HangyuanLiu — APPROVED** [review]

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by xlwh** [comment]
run starrocks_fe_unittest

**Comment by xlwh** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 103 / 127 (81.10%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|3|4|75.00%|[302]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowAlterStmtAnalyzer.java|67|87|77.01%|[32, 43, 44, 74, 75, 89, 108, 114, 115, 126, 131, 137, 143, 149, 150, 153, 161, 166, 170, 171]|
|:large_blue_circle:|com/starrocks/analysis/ShowAlterStmt.java|14|16|87.50%|[121, 178]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|16|17|94.12%|[576]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|2|2|100.00%|[]|

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**